Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -527,6 +530,106 @@ class Gx2Fitter {
 
           // We count the processed state
           ++result.processedStates;
+
+          // Update the number of holes count only when encountering a
+          // measurement
+          result.measurementHoles = result.missedActiveSurfaces.size();
+        } else if (surface->associatedDetectorElement() != nullptr ||
+                   surface->surfaceMaterial() != nullptr) {
+          // Here we handle material and holes
+          // TODO add material handling
+          ACTS_VERBOSE("Non-Measurement surface " << surface->geometryId()
+                                                  << " detected.");
+
+          // We only create track states here if there is already a measurement
+          // detected or if the surface has material (no holes before the first
+          // measurement)
+          if (result.measurementStates > 0
+              // || surface->surfaceMaterial() != nullptr
+          ) {
+            ACTS_VERBOSE("Handle hole.");
+
+            auto& fittedStates = *result.fittedStates;
+
+            // Mask for the track states. We don't need Smoothed and Filtered
+            TrackStatePropMask mask =
+                ~(TrackStatePropMask::Smoothed | TrackStatePropMask::Filtered);
```

Comment:
making the turned on flags explicit might be more digestible

Reply:
I changed them everywhere :)
https://github.com/acts-project/acts/pull/3138